consul/resource_consul_config_entry_v2_exported_services.go

```diff
@@ -0,0 +1,205 @@
+		Read:   resourceConsulV2ExportedServicesRead,
+		Delete: resourceConsulV2ExportedServicesDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the config entry to read.",
+			},
+
+			"kind": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The kind of exported services config (ExportedServices, NamespaceExportedServices, PartitionExportedServices).",
+			},
+
+			"partition": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The partition the config entry is associated with.",
+				ForceNew:    true,
+			},
+
+			"namespace": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The namespace the config entry is associated with.",
+				ForceNew:    true,
+			},
+
+			"services": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "The exported services.",
+			},
+
+			"partition_consumers": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "The exported service partition consumers.",
+			},
+			"peer_consumers": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "The exported service peer consumers.",
+			},
+			"sameness_group_consumers": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "The exported service sameness group consumers.",
+			},
+		},
+	}
+}
+
+func resourceConsulV2ExportedServicesCreate(d *schema.ResourceData, meta interface{}) error {
+	return resourceConsulV2ExportedServicesUpdate(d, meta)
+}
+
+func resourceConsulV2ExportedServicesUpdate(d *schema.ResourceData, meta interface{}) error {
+	client, _, wOpts := getClient(d, meta)
+	name := d.Get("name").(string)
+	kind := d.Get("kind").(string)
+	gvk := &GVK{
+		Group:   "multicluster",
+		Version: "v2",
+		Kind:    kind,
+	}
+	var consumers []map[string]any
+	peerConsumers := d.Get("peer_consumers").([]interface{})
+	for _, p := range peerConsumers {
+		consumers = append(consumers, map[string]any{"peer": p})
+	}
+	partitionConsumers := d.Get("partition_consumers").([]interface{})
+	for _, ap := range partitionConsumers {
+		consumers = append(consumers, map[string]any{"partition": ap})
+	}
+	samenessConsumers := d.Get("sameness_group_consumers").([]interface{})
+	for _, sg := range samenessConsumers {
+		consumers = append(consumers, map[string]any{"sameness_group": sg})
+	}
+	data := map[string]any{"consumers": consumers}
+	services := d.Get("services").([]interface{})
+	if len(services) > 0 {
+		data["services"] = services
+	}
```

Comment:
Should we return if `data["services"]` is blank?

Reply:
no, because NamespaceExportedServices and PartitionExportedServices do not explicitly set services